exonum-java-binding/core/rust/src/runtime/service_factory_adapter.rs

```diff
@@ -0,0 +1,103 @@
+// Copyright 2019 The Exonum Team
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use exonum::{
+    blockchain::Service,
+    helpers::fabric::{Command, CommandExtension, Context, ServiceFactory},
+};
+use runtime::{cmd::Run, config::Config, java_service_runtime::JavaServiceRuntime};
+use std::sync::{Once, ONCE_INIT};
+
+static mut JAVA_SERVICE_RUNTIME: Option<JavaServiceRuntime> = None;
+static JAVA_SERVICE_RUNTIME_INIT: Once = ONCE_INIT;
+static JAVA_SERVICE_CREATE_COMMAND: Once = ONCE_INIT;
+
+/// Adapts current single-service interface of Exonum to multiple EJB services until dynamic
+/// services are implemented. Tracks the `JavaServiceRuntime` instantiation and command extension
+/// and represents factory for every user service relying on current `ServiceFactory` interface.
+pub struct JavaServiceFactoryAdapter {
+    name: String,
+    artifact_path: String,
+}
+
+impl JavaServiceFactoryAdapter {
+    /// Creates new instance with given service name and path to the artifact.
+    pub fn new<S: AsRef<str>>(name: S, artifact_path: S) -> Self {
```

Comment:
Keep in mind, that such signature requires both parameters to have the exact same type.

Reply:
Yeap, re-written to strings